wled00/FX.cpp

```diff
@@ -4861,6 +4861,53 @@ static const char _data_FX_MODE_FLOWSTRIPE[] PROGMEM = "Flow Stripe@Hue speed,Ef
 #define XY(x,y) SEGMENT.XY(x,y)
 
 
+/////////////////////////////
+//    2D Analog Clock      //
+/////////////////////////////
+uint16_t mode_2DAnalogClock(void) {                  // By Andras Fekete (bandi13)
+  if (!strip.isMatrix || !SEGMENT.is2D()) return mode_static(); // not a 2D set-up
+
+  const int cols = SEGMENT.virtualWidth();
+  const int rows = SEGMENT.virtualHeight();
+  const int centerX = (cols-!(cols%2)) >> 1;  // use odd sized circle
+  const int centerY = (rows-!(rows%2)) >> 1;  // use odd sized circle
+  const int radius = min(centerX, centerY);
+  const bool soft = radius > 6 && SEGMENT.check2;
+
+  SEGMENT.fadeToBlackBy(240);
+  SEGMENT.drawCircle(centerX, centerY, radius, DARKGREY, soft);
+  if (radius > 8) SEGMENT.drawCircle(centerX, centerY, radius-1, DARKGREY, soft); // thicker circle
+
+  time_t hours_in_day = (localTime / (60 * 60)) % 12;
+  time_t minutes_in_day = (localTime / (60)) % 60;
+  time_t seconds_in_day = (localTime) % 60;
+
+  float hour_angle = radians(30.0f * (hours_in_day + minutes_in_day / 60.0f) - 90.0f);
+  int hour_len = radius / 2;
+  int hour_x = centerX + hour_len * cos_t(hour_angle);
+  int hour_y = centerY + hour_len * sin_t(hour_angle);
+
+  float minute_angle = radians(6.0f * (minutes_in_day + seconds_in_day / 60.0f) - 90.0f);
+  int minute_len = (radius * 7) / 10;
+  int minute_x = centerX + minute_len * cos_t(minute_angle);
+  int minute_y = centerY + minute_len * sin_t(minute_angle);
+
+  if (SEGMENT.check1) {
+    float second_angle = radians(6.0f * seconds_in_day - 90.0f);
+    int second_len = (radius * 9) / 10;
+    int second_x = centerX + second_len * cos_t(second_angle);
+    int second_y = centerY + second_len * sin_t(second_angle);
+    SEGMENT.drawLine(centerX, centerY, second_x, second_y, SEGCOLOR(2), soft);
+  }
+
+  SEGMENT.drawLine(centerX, centerY, minute_x, minute_y, SEGCOLOR(1), soft);
+  SEGMENT.drawLine(centerX, centerY, hour_x, hour_y, SEGCOLOR(0), soft);
+
+  return FRAMETIME;
+} // mode_2DAnalogClock()
+static const char _data_FX_MODE_2DANALOGCLOCK[] PROGMEM = "Analog Clock@,,,,,Seconds,Soft;Hour,Minute,Second;;2;o1=1,o2=1";
```

Comment:
Any way to make the default colors actually correct? On startup, Hour=white, Minute=black, Second=black for me.

Reply:
Unfortunately you can't but I guess you are *not* using *every* effect using *default* values. Or are you?
If you are not, you should consider using presets which save every configuration item.

Reply:
Effect metadata format is documented here: 
https://kno.wled.ge/interfaces/json-api/#effect-metadata

It looks like you can define default colors in metadata, too.

Reply:
The default values should be ones that work for most cases out of the box. When you buy a car/phone/computer/anything it should work out of the box without you having to edit parameters to make it functional. Note that I didn't say customized, just functional out of the gate.

Maybe you could have a special case, that if the Hour=White and Minute=Second=Black that it would set the colors to Red, Green, Blue respectively?

Reply:
When you buy it, yes. If you get it for free, it is just a nice touch.

> Maybe you could have a special case, that if the Hour=White and Minute=Second=Black that it would set the colors to Red, Green, Blue respectively?

You can do that in effect initialisation code that starts as:

```c++
if (SEGENV.call == 0) {
  SEGMENT.colors[0] = <hour_color>;
...
}
```

But I would only change those if they a black.
There are examples elsewhere in the code.

The other option is to enhance metadata. That will be a lot more work IMO.